artemis/src/test/java/tech/pegasys/artemis/cli/util/YamlConfigFileDefaultProviderTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.cli.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.yaml.snakeyaml.Yaml;
+import picocli.CommandLine;
+
+class YamlConfigFileDefaultProviderTest {
+
+  @Test
+  void parsingValidYamlFilePopulatesCommandObject(@TempDir final Path tempDir) throws IOException {
+    final Path configFile = writeToYamlConfigFile(defaultOptions(), tempDir);
+    final CommandLine commandLine = new CommandLine(TestCommand.class);
+    commandLine.setDefaultValueProvider(
+        new YamlConfigFileDefaultProvider(commandLine, configFile.toFile()));
+    commandLine.parseArgs();
+    final TestCommand testCommand = commandLine.getCommand();
+
+    Assertions.assertThat(testCommand.getCount()).isEqualTo(10);
+    Assertions.assertThat(testCommand.getNames()).containsExactlyInAnyOrder("a", "b");
+    Assertions.assertThat(testCommand.isTestEnabled()).isTrue();
+  }
+
+  @Test
+  void parsingEmptyConfigFileThrowsException(@TempDir final Path tempDir) throws IOException {
+    final Path configFile = writeToYamlConfigFile(Collections.emptyMap(), tempDir);
+    final CommandLine commandLine = new CommandLine(TestCommand.class);
+    commandLine.setDefaultValueProvider(
+        new YamlConfigFileDefaultProvider(commandLine, configFile.toFile()));
+
+    Assertions.assertThatExceptionOfType(CommandLine.ParameterException.class)
+        .isThrownBy(() -> commandLine.parseArgs())
```

Comment:
nit: All of these can just use method references
```suggestion
        .isThrownBy(commandLine::parseArgs)
```

Reply:
👍 using method references.